imap_processing/codice/utils/codice_utils.py

```diff
@@ -15,3 +67,77 @@ class CoDICECompression(IntEnum):
     LOSSLESS = 4
     LOSSY_A_LOSSLESS = 5
     LOSSY_B_LOSSLESS = 6
+
+
+def add_metadata_to_array(packet, metadata_arrays):
+    """Add metadata to the metadata_arrays.
+
+    Parameters
+    ----------
+    packet : space_packet_parser.parser.Packet
+        CODICE data packet
+    metadata_arrays : dict
+        Metadata arrays
+    """
+    for key, value in packet.header.items():
+        metadata_arrays.setdefault(key, []).append(value.raw_value)
+
+    for key, value in packet.data.items():
+        if key in CONVERT_TO_EU_LIST:
+            # TODO: Convert value to engineering units
+            # For now, just use the raw value
+            converted_value = value.raw_value
+            metadata_arrays.setdefault(key, []).append(converted_value)
+        else:
+            metadata_arrays.setdefault(key, []).append(value.raw_value)
```

Comment:
You might want to bring at least the `metadata_arrays.setdefault(key, []).append(...)` block outside of the if blocks since that is essentially repeated and only update your value of interest within the choice.
```suggestion
        # TODO: Convert value to engineering units
        # For now, just use the raw value
        converted_value = value.raw_value if key in CONVERT_TO_EU_LIST else value.raw_value
        metadata_arrays.setdefault(key, []).append(converted_value)
```